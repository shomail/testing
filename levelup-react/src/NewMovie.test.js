import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import NewMovie from './NewMovie';

afterEach(cleanup); // cleanup the rendered DOM tree after test ran

test('<NewMovie />', () => {
  // r-t-l defaults to full on render of the component tree
  // in r-t-l you don't do shallow renders
  // because our users don't care about it so do we
  // instead enzyme recommends shallow render of comp
  // here in this case MovieForm comp also gets rendered

  const {
 getByTestId, debug, queryByTestId, container, getByText 
} = render(<NewMovie />);
  expect(getByTestId('page-title').textContent).toBe('Enter New Movie');
  expect(queryByTestId('movie-form')).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();

  fireEvent.click(getByText('Save'));
});
