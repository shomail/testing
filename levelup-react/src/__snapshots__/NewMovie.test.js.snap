// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie /> 1`] = `
<div>
  <h1
    data-testid="page-title"
  >
    Enter New Movie
  </h1>
  <div>
    <form
      data-testid="movie-form"
    >
      <input
        type="text"
      />
      <button
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
